test/users-query-test.ts

```diff
@@ -0,0 +1,176 @@
+import { expect } from 'chai';
+import { USERS_QUERY } from './graphql/queries';
+import { seed } from '../src/seed/seedaddress';
+import jwt from 'jsonwebtoken';
+import axios from 'axios';
+import prisma from '../src/prisma';
+import { User } from '../src/types/types';
+
+describe('Users Query', function () {
+  let authToken: string;
+  let users: User[];
+
+  this.beforeAll(async () => {
+    users = await seed();
+  });
+
+  beforeEach(async () => {
+    authToken = jwt.sign({ userID: 1 }, process.env.JWT_SECRET as string, { expiresIn: '1h' });
+  });
+
+  this.afterAll(async () => {
+    await prisma.$disconnect();
+  });
+
+  it('should return paginated users successfully', async () => {
+    const limit = 10;
+    const offset = 0;
+    const queryUsers = USERS_QUERY({ limit, offset });
+
+    const headers = { Authorization: `Bearer ${authToken}` };
+    const response = await axios.post('http://localhost:4000', queryUsers, { headers });
+
+    const returnedUsers = response.data.data.Users;
+
+    expect(returnedUsers.users.length).to.eq(10);
+    expect(returnedUsers.totalUsers).to.eq(users.length);
+    expect(returnedUsers.hasPreviousPage).to.eq(false);
+    expect(returnedUsers.hasNextPage).to.eq(true);
+
+    for (let i = 0; i < limit; i++) {
+      expect(returnedUsers.users[i]).to.deep.include({
+        id: users[i + offset].id.toString(),
+        name: users[i + offset].name,
+        email: users[i + offset].email,
+        birthDate: users[i + offset].birthDate.getTime().toString(),
+        addresses: users[i + offset].addresses.map((address) => ({
+          ...address,
+          id: address.id.toString(),
+          userID: address.userID.toString(),
+        })),
+      });
+    }
```

Comment:
Tá ok assim, mas acho que daria pra usar to be deep eq em toda a resposta, usando `.map` pra mapear os expectedUsers.